hibernate-core/src/test/java/org/hibernate/jpa/test/criteria/components/ComponentInWhereClauseTest.java

```diff
@@ -0,0 +1,351 @@
+									 Root<Employee> root = query.from( Employee.class );
+
+									 query.where( root.get( "projects" )
+														  .get( "currentProject" )
+														  .in( projects.getCurrentProject() ) );
+
+									 final List<Employee> results = entityManager.createQuery( query ).getResultList();
+									 assertThat( results.size(), is( 1 ) );
+								 }
+		);
+	}
+
+	@MappedSuperclass
+	public static abstract class AbstractEntity {
+		@Id
+		@GeneratedValue(strategy = GenerationType.AUTO)
+		protected Long id;
+
+		public Long getId() {
+			return id;
+		}
+	}
+
+	@Entity(name = "Employee")
+	@Table(name = "EMPLOYEE")
+	public static class Employee extends AbstractEntity {
+
+		@Embedded
+		private Projects projects;
+
+		@Embedded
+		private ContactDetail contactDetail;
+
+		public void setProjects(Projects projects) {
+			this.projects = projects;
+		}
+
+		public void setContactDetail(ContactDetail contactDetail) {
+			this.contactDetail = contactDetail;
+		}
+	}
+
+	@Embeddable
+	public static class ContactDetail {
+		private String email;
+
+		@ElementCollection
+		private List<Phone> phones = new ArrayList<>();
+
+		public void addPhone(Phone phone) {
+			this.phones.add( phone );
+		}
+
+		public String getEmail() {
+			return email;
+		}
+
+		public void setEmail(String email) {
+			this.email = email;
+		}
+	}
+
+	@Embeddable
+	public static class Projects {
+
+		@OneToMany(cascade = CascadeType.PERSIST)
+		private Set<Project> previousProjects = new HashSet<>();
+
+		@ManyToOne(cascade = CascadeType.PERSIST)
+		private Project currentProject;
+
+		public void setCurrentProject(Project project) {
+			this.currentProject = project;
+		}
+
+		public void addPreviousProject(Project project) {
+			this.previousProjects.add( project );
+		}
+
+		public Set<Project> getPreviousProjects() {
+			return previousProjects;
+		}
+
+		public Project getCurrentProject() {
+			return currentProject;
+		}
+	}
+
+	@Entity(name = "Project")
+	@Table(name = "PROJECT")
+	public static class Project extends AbstractEntity {
+
+		public Project() {
+		}
+
+		public Project(String name) {
+			this.name = name;
+		}
+
+		private String name;
+
+		public void setName(String name) {
+			this.name = name;
+		}
+
+		public String getName() {
+			return this.name;
+		}
+	}
+
+	@Embeddable
+	public static class Phone {
+		private String number;
+
+		public Phone() {
+		}
+
+		public Phone(String number) {
+			this.number = number;
+		}
+
+		public String getNumber() {
+			return this.number;
+		}
+	}
+
+	@Entity(name = "Person")
+	@Table(name="PERSON")
+	public static class Person extends AbstractEntity {
+		@Embedded
+		private Information information;
+
+		public Information getInformation() {
+			return information;
+		}
+
+		public void setInformation(Information information) {
+			this.information = information;
+		}
+	}
+
+	@Embeddable
+	public static class Information {
+		@Embedded
+		private ContactDetail infoContactDetail;
+
```

Comment:
I changed Information#contactDetail to #infoContactDetail, so the attribute name would be different from Employee#contactDetail.